pkg/cable/libreswan/libreswan.go

```diff
@@ -24,8 +27,19 @@ const (
 	cableDriverName = "libreswan"
 )
 
+// The metrics are gauges because we want to set the absolute value
```

Comment:
I think we should not expose cable-driver specific gauges, 

Ideally we should create an interface on the cable-drivers, so we can expose tx_bytes / rx_bytes per remote endpoint, we could label with the cable driver may be.


I think it's OK if currently until we create such interface we publish with a generic name. rx_bytes, tx_bytes.  But I'd expected this to be per remote cluster endpoint, instead of a global counter.

Reply:
I’m following the rule that metrics should be gathered as close as possible to the point of origin. The idea here is to provide metrics to see how they can be integrated, and how they compare with those from WireGuard once we get them.

Ultimately I think the common interface may come from the metrics themselves, not necessarily the cable driver interface. That is to say, we should define the metrics we want, and then let the cable drivers implement them.

And yes, this should be per remote cluster endpoint, with appropriate labels...

Reply:
Yes, but... I still believe it makes more sense to keep a single name
rx_bytes, tx_bytes, regardless of the driver

One driver will emit metrics with one label, other with a different label. Can that be done?

Reply:
Yes, that’s the plan.